File: core/src/main/java/com/graphhopper/routing/ev/TurnCost.java
package com.graphhopper.routing.ev;

import com.graphhopper.util.BitUtil;

import static com.graphhopper.routing.util.EncodingManager.getKey;

public class TurnCost {

    public static String key(String prefix) {
        return getKey(prefix, "turn_cost");
    }

    /**
     * This creates an EncodedValue specifically for the turn costs
     */
    public static DecimalEncodedValue create(String name, int maxTurnCosts) {
        int turnBits = BitUtil.countBitValue(maxTurnCosts);
        return new DecimalEncodedValueImpl(key(name), turnBits, 0, 1, false, false, true);
    }
}

File: core/src/main/kotlin/com/graphhopper/routing/ev/TurnCost.kt
package com.graphhopper.routing.ev

import com.graphhopper.routing.util.EncodingManager.Companion.getKey
import com.graphhopper.util.BitUtil

object TurnCost {

    @JvmStatic
    fun key(prefix: String): String = getKey(prefix, "turn_cost")

    /**
     * This creates an EncodedValue specifically for the turn costs
     */
    @JvmStatic
    fun create(name: String, maxTurnCosts: Int): DecimalEncodedValue {
        val turnBits = BitUtil.countBitValue(maxTurnCosts)
        return DecimalEncodedValueImpl(key(name), turnBits, 0.0, 1.0, false, false, true)
    }
}
